vmdk_plugin/plugin.go

```diff
@@ -208,11 +209,50 @@ func (d *vmdkDriver) unmountVolume(name string) error {
 // (until Mount is called).
 // Name and driver specific options passed through to the ESX host
 func (d *vmdkDriver) Create(r volume.Request) volume.Response {
-	err := d.ops.Create(r.Name, r.Options)
+	// Defaults to ext4 if not specified
+	if _, ok := r.Options["fstype"]; ok == false {
+		r.Options["fstype"] = "ext4"
+	}
+
+	// Verify the existance of filesystem tools
+	mkfscmd := "/sbin/mkfs." + r.Options["fstype"]
+	_, err := os.Lstat(mkfscmd)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Not found ")
+		return volume.Response{Err: err.Error()}
+	}
+
+	err = d.ops.Create(r.Name, r.Options)
 	if err != nil {
 		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Create volume failed ")
 		return volume.Response{Err: err.Error()}
 	}
+
+	// Handle filesystem creation
+	dev, err := d.ops.Attach(r.Name, nil)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Attach volume failed ")
+		return volume.Response{Err: err.Error()}
+	}
+
+	device, err := fs.GetDevicePath(dev)
+	if err != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": err}).Error("Could not find attached device ")
+		return volume.Response{Err: err.Error()}
+	}
+
+	err = fs.Mkfs(mkfscmd, r.Name, device)
```

Comment:
this could be very slow for large disk sizes.
Ideally it should be a go func with some timeout for ("in progress") messages, and killable but at the very least need we need to log start/end.
Note: the current code does not handle large disk formats very well either.